infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/devfile/KubernetesComponentToWorkspaceApplier.java

```diff
@@ -215,23 +215,62 @@ private void applyProjectsVolumes(List<PodData> podsData, List<HasMetadata> comp
    * attribute set.
    */
   private Map<String, MachineConfigImpl> prepareMachineConfigs(
-      List<PodData> podsData, @Nullable String componentAlias) {
+      List<PodData> podsData, ComponentImpl component) throws DevfileException {
     Map<String, MachineConfigImpl> machineConfigs = new HashMap<>();
     for (PodData podData : podsData) {
       for (Container container : podData.getSpec().getContainers()) {
         String machineName = machineName(podData, container);
 
         MachineConfigImpl config = new MachineConfigImpl();
-        if (!isNullOrEmpty(componentAlias)) {
-          config.getAttributes().put(DEVFILE_COMPONENT_ALIAS_ATTRIBUTE, componentAlias);
+        if (!isNullOrEmpty(component.getAlias())) {
+          config.getAttributes().put(DEVFILE_COMPONENT_ALIAS_ATTRIBUTE, component.getAlias());
         }
-
+        provisionVolumes(component, container, config);
         machineConfigs.put(machineName, config);
       }
     }
     return machineConfigs;
   }
 
+  private void provisionVolumes(
+      ComponentImpl component, Container container, MachineConfigImpl config)
+      throws DevfileException {
+    for (org.eclipse.che.api.workspace.server.model.impl.devfile.VolumeImpl v :
```

Comment:
can we rename the variable to for example `componentVolume`?

Reply:
I would prefer to use more descriptive names in bigger blocks, and shortcuts for lambdas